pkg/controller/volume/persistentvolume/binder_test.go

```diff
@@ -362,6 +362,38 @@ func TestSync(t *testing.T) {
 			newClaimArray("claim2-10", "uid2-10", "1Gi", "volume2-10", v1.ClaimPending, nil),
 			[]string{"Warning VolumeMismatch"}, noerrors, testSyncClaim,
 		},
+		{
+			// syncClaim with claim pre-bound to a released PV that is pre-bound to the
+			// claim by UID. Check it gets bound and no pvutil.AnnBoundByController is
+			// set.
+			"2-11 - claim prebound to released and prebound volume by UID",
+			newVolumeArray("volume2-5", "1Gi", "uid2-5", "claim2-5", v1.VolumeReleased, v1.PersistentVolumeReclaimRetain, classEmpty),
+			newVolumeArray("volume2-5", "1Gi", "uid2-5", "claim2-5", v1.VolumeBound, v1.PersistentVolumeReclaimRetain, classEmpty),
+			newClaimArray("claim2-5", "uid2-5", "1Gi", "volume2-5", v1.ClaimPending, nil),
+			newClaimArray("claim2-5", "uid2-5", "1Gi", "volume2-5", v1.ClaimBound, nil, pvutil.AnnBindCompleted),
+			noevents, noerrors, testSyncClaim,
+		},
+		{
+			// syncClaim with claim pre-bound to a released PV that is pre-bound to the
+			// claim by name. Check it gets bound and no pvutil.AnnBoundByController is
+			// set.
+			"2-12 - claim prebound to released and prebound volume by name",
+			newVolumeArray("volume2-5", "1Gi", "", "claim2-5", v1.VolumeReleased, v1.PersistentVolumeReclaimRetain, classEmpty),
```

Comment:
This enters a gray area: under normal circumstances, a volume can't enter Released phase if it has no PVC UID. I know that PV controller ignores the Phase, so the volume looks Pending/Available and it binds the volume to PVC, however, I am not sure we should codify this behavior in unit tests. Does it cover any code path not covered by previous unit tests?

Reply:
what about this scenario?

- PV enters into Released state when associated PVC is deleted
-  User pre-binds the PV to the PVC `claim2-5` by setting `{Name: "claim2-5"}` to `claimRef` field
  - no UID, but PV is in the Released state
- Users create a PVC named `claim2-5`

Reply:
> Does it cover any code path not covered by previous unit tests?

This scenario is meaningful IMO. But I agree it does not cover any new code path. Because the PV controller ignores the phase, it's already covered by `"2-4 - claim prebound to prebound volume by name"` except the PV state is Available.

Reply:
> old claimRef is removed, so no UID, but PV is in the Released state

This is possible, but PV controller will fix the status to Available in the next sync.